src/api.py

```diff
@@ -0,0 +1,44 @@
+import hashlib
+import requests
+
+class Response:
+    def __init__(self, filename, type_of_data, size, url, checksum):
+        self.filename = filename
+        self.type_of_data = type_of_data
+        self.size=size
+        self.url=url
+        self.checksum=checksum
+
+
+def generate_filename(url):
+    """
+    split url on basis of '/' and start iterating from end till you find non empty value
+    """
+    for i in range(len(url.split('/'))-1):
+        name = url.split('/')[-(i+1)]
+        if not(name =='' ):
+            break
+    return name
+
+
+def fetch(url,location):
+    # Fetching data from URL
```

Comment:
What is the value of this comment? I think we can dispense of it. Instead you should have a descriptive docstring that explains the returned value.